code/modules/reagents/reagent_containers/food/canned_food.dm

```diff
@@ -0,0 +1,171 @@
+//Canned Foods - crack open, eat.
+
+#define OPEN_HARD 1
+#define OPEN_EASY 0
+
+/obj/item/chems/food/snacks/can
+	name = "void can"
+	icon = 'icons/obj/food_canned.dmi'
+	atom_flags = 0
+
+	var/sealed = TRUE
+	var/open_complexity = OPEN_HARD
+
+/obj/item/chems/food/snacks/can/Initialize()
+	. = ..()
+	if(!sealed)
+		unseal()
+
+/obj/item/chems/food/snacks/can/examine(mob/user)
+	. = ..()
+	to_chat(user, "It is [sealed ? "" : "un"]sealed.")
+	to_chat(user, "It looks [open_complexity ? "hard" : "easy "] to open.")
+
+/obj/item/chems/food/snacks/can/proc/unseal(mob/user)
+	playsound(src, 'sound/effects/canopen.ogg', rand(10, 50), 1)
+	atom_flags |= ATOM_FLAG_OPEN_CONTAINER
+	sealed = FALSE
+
+/obj/item/chems/food/snacks/can/attack(mob/M, mob/user, def_zone)
+	if(force && !(obj_flags & ITEM_FLAG_NO_BLUDGEON) && user.a_intent == I_HURT)
+		return ..()
+
+	if(standard_feed_mob(user, M))
+		update_icon(src)
+		return
+
+	return FALSE
+
+/obj/item/chems/food/snacks/can/standard_feed_mob(mob/user, mob/target)
+	if(!ATOM_IS_OPEN_CONTAINER(src))
+		to_chat(user, SPAN_NOTICE("You need to open \the [src] first!"))
+		return TRUE
+	return ..()
+
+/obj/item/chems/food/snacks/can/attack_self(mob/user)
+	if(!ATOM_IS_OPEN_CONTAINER(src) && sealed && !open_complexity)
+		to_chat(user, SPAN_NOTICE("You unseal \the [src] with a crack of metal."))
+		unseal()
+
+/obj/item/chems/food/snacks/can/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/knife) && sealed)
```

Comment:
Isn't there a fancy system for tools like this now?

Reply:
Knives aren't considered tools currently.